### PR TITLE
Wito/feat/add resource sets

### PR DESCRIPTION
**Pull request by WitoDelnat**
This PR adds all resource set types.

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/components/organisms/CompareModal/CompareModal.tsx [hunk 1]
```diff
@@ -34,17 +35,24 @@ export const DiffModal: React.FC<Props> = ({visible, onClose}) => {
...
+        </Col>
+      </Row>
 
+      <div
+        style={{
+          height: `calc(100% - ${height}px - 66px)`,
+          position: 'relative',
+          overflowY: 'auto',
+          overflowX: 'hidden',
+        }}
```
Comment (WitoDelnat) [inline src/components/organisms/CompareModal/CompareModal.tsx:54]: Please ignore all these `style` and layout things while reviewing. I'll come back and fix this properly later I add more functionality such as different operations and filter.

File: src/redux/reducers/compare.ts [hunk 2]
```diff
@@ -359,27 +407,52 @@ export const selectComparisonListItems = createSelector(
...
+      predicate: (action, state) => {
         const resourceSetUpdated = isAnyOf(resourceSetSelected, resourceSetRefreshed, resourceSetCleared);
         if (!resourceSetUpdated(action)) return false;
+
         const actionSide = action.payload.side;
-        return actionSide === side;
+        if (actionSide !== side) return false;
+
+        const resourceSet = selectResourceSet(state.compare, side);
+        if (!isCompleteResourceSet(resourceSet)) return false;
```
Comment (WitoDelnat) [inline src/redux/reducers/compare.ts:443]: So this is the whole essence of allowing this kind of user experience.

The configuration in our store is kept partially and only once it's completely present do we move on to the next stage.

File: src/redux/services/compare/fetchResources.ts [hunk 3]
```diff
@@ -56,7 +70,78 @@ async function fetchResourcesFromCluster(state: RootState): Promise<K8sResource[
     const allYaml = fulfilledResults.map(r => (r as any).value).join(YAML_DOCUMENT_DELIMITER_NEW_LINE);
     return extractK8sResources(allYaml, CLUSTER_DIFF_PREFIX + String(kc.currentContext));
   } catch (err) {
-    log.debug('fetch resources form cluster failed', err);
+    log.debug('fetch resources from cluster failed', err);
+    throw err;
+  }
+}
+
+async function previewHelmResources(state: RootState, options: HelmResourceSet): Promise<K8sResource[]> {
```
Comment (WitoDelnat) [inline src/redux/services/compare/fetchResources.ts:78]: I wonder how we can re-use this better.. I think the `thunk` and this `listener` should both use one underlying abstraction to fetch resources though it felt a bit out of scope for this. Wdyt?

Comment (devcatalin) [inline src/redux/services/compare/fetchResources.ts:78]: we'll probably have to see if we want/can reuse some of this code

**Review by devcatalin — APPROVED** [review]

File: src/components/organisms/CompareModal/ResourceSetSelector.tsx [hunk 4]
```diff
@@ -75,3 +68,154 @@ export const ResourceSetSelector: React.FC<Props> = ({side}: Props) => {
     </S.ResourceSetSelectorDiv>
   );
 };
+
+function ResourceSetTypeSelect({side}: Props) {
```
Comment (WitoDelnat) [inline src/components/organisms/CompareModal/ResourceSetSelector.tsx:72]: Shall I extract all these selectors to separate files? At that point I feel like the ResourceSetSelector should become its own directory.

Comment (devcatalin) [reply src/components/organisms/CompareModal/ResourceSetSelector.tsx:72]: yeah, I think that makes sense

**Pull request merged**